source/state_representation/include/state_representation/State.hpp

```diff
@@ -48,6 +48,11 @@ class State : public std::enable_shared_from_this<State> {
    */
   State(const State& state);
 
+  /**
+   * @brief Virtual destructor
+   */
+  virtual ~State() = default;
```

Comment:
I'm surprised this went missing for so long. Thanks for the incremental improvement here.